### PR TITLE
add sanitizer function

### PR DESCRIPTION
**Pull request by jnewb1**

### CODE REVIEW

File: selfdrive/test/sanetize.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Utilities for sanetizing routes of only essential data
+
+from openpilot.tools.lib.logreader import LogIterable
+
+PRESERVE_SERVICES = ["can", "carParams", "pandaStates", "pandaStateDEPRECATED"]
+
+def sanetize(lr: LogIterable) -> LogIterable:
```
Comment (sshane) [inline selfdrive/test/sanetize.py:7]:
```suggestion
def sanitize(lr: LogIterable) -> LogIterable:
```

File: selfdrive/test/sanetize.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Utilities for sanetizing routes of only essential data
```
Comment (sshane) [inline selfdrive/test/sanetize.py:1]:
```suggestion
# Utilities for sanitizing routes of only essential data
```

**Comment by sshane** [comment]
you can work in https://github.com/commaai/openpilot/pull/30545 or close if you plan on splitting it up

**Comment by jnewb1** [comment]
> you can work in https://github.com/commaai/openpilot/pull/30545 or close it if you plan on splitting it up

still deciding the end goal for the segments in https://github.com/commaai/openpilot/pull/30891 so I will start spliting it up

File: selfdrive/test/helpers.py [hunk 3]
```diff
@@ -79,3 +80,11 @@ def read_segment_list(segment_list_path):
...
 
   return [(platform[2:], segment) for platform, segment in zip(seg_list[::2], seg_list[1::2], strict=True)]
+
+
+# Utilities for sanitizing routes of only essential data for testing car ports and doing validation.
+
+PRESERVE_SERVICES = ["can", "carParams", "pandaStates", "pandaStateDEPRECATED"]
+
+def sanitize(lr: LogIterable) -> LogIterable:
+  return filter(lambda msg: msg.which() in PRESERVE_SERVICES, lr)
```
Comment (sshane) [inline selfdrive/test/helpers.py:90]: Always two spaces around top level functions.

Comment (sshane) [reply selfdrive/test/helpers.py:90]: Also, you probably want to zero out the carVin inside carParams

Comment (jnewb1) [reply selfdrive/test/helpers.py:90]: maybe just the last few digits (the serial number)? we might want to use the vin for fingerprinting on some brands

Comment (sshane) [reply selfdrive/test/helpers.py:90]: that's fine, I'm hiding the last 8 on the FP bot

**Pull request merged**